### PR TITLE
Classes Shape, Circle, Triangle and Square

### PR DESCRIPTION
**Pull request by sonyashnik**
Homework

### CODE REVIEW

File: csharp/HW.02.Task1/Shape.cs [hunk 1]
```diff
@@ -0,0 +1,65 @@
+﻿public class Shape
```
Comment (VadzimPapko) [inline csharp/HW.02.Task1/Shape.cs]: replace to abstract class

File: csharp/HW.02.Task1/Shape.cs [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+{
+    protected  string _name { get; set; }
+
+    public string Name
+    {
+        get { return _name; }
+        set { _name = value; }
+    }
```
Comment (VadzimPapko) [inline csharp/HW.02.Task1/Shape.cs]: replace to auto property `public string Name {get;set;}`

File: csharp/HW.02.Task1/Shape.cs [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+    public virtual double GetSquare()
+    {
+        return 0;
```
Comment (VadzimPapko) [inline csharp/HW.02.Task1/Shape.cs]: default

File: csharp/HW.02.Task1/Shape.cs [hunk 4]
```diff
@@ -0,0 +1,65 @@
+﻿public class Shape
+{
+    protected  string _name { get; set; }
+
+    public string Name
+    {
+        get { return _name; }
+        set { _name = value; }
+    }
+    public virtual double GetSquare()
```
Comment (VadzimPapko) [inline csharp/HW.02.Task1/Shape.cs]: add empty line above

File: csharp/HW.02.Task1/Shape.cs [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+    {
+        return 0;
+    }
+    public string ShapeInfo()
```
Comment (VadzimPapko) [inline csharp/HW.02.Task1/Shape.cs]: add empty line above

File: csharp/HW.02.Task1/Shape.cs [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+    {
+        return $"This is {this.Name}. CLR Type is {this.GetType()}. Square is {this.GetSquare()}.";
+    }
+}
+public class Square : Shape
+{
+    private int Side { get; set; }
```
Comment (VadzimPapko) [inline csharp/HW.02.Task1/Shape.cs]: replace to field

File: csharp/HW.02.Task1/Shape.cs [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+    public Square(int s)
```
Comment (VadzimPapko) [inline csharp/HW.02.Task1/Shape.cs]: rename operator `s` to human readable name

File: csharp/HW.02.Task1/Shape.cs [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+    {
+        Side = s;
+        this.Name = "square";
```
Comment (VadzimPapko) [inline csharp/HW.02.Task1/Shape.cs]: simplify `this`

File: csharp/HW.02.Task1/Shape.cs [hunk 9]
```diff
@@ -0,0 +1,65 @@
...
+    public override double GetSquare()
+    {
+        return Math.Pow(Side, 2);
+    }
+}
+
+public class Circle : Shape
+{
+    private int Radius { get; set; }
+  public Circle (int @radius)
```
Comment (VadzimPapko) [inline csharp/HW.02.Task1/Shape.cs]: format issue

File: csharp/HW.02.Task1/Shape.cs [hunk 10]
```diff
@@ -0,0 +1,65 @@
...
+    {
+        Radius = @radius;
+        this.Name = "circle";
+    }
+
+    public override double GetSquare()
+    {
+        return Math.Round(Math.PI * Math.Pow(Radius, 2), 2);
+    }
+
```
Comment (VadzimPapko) [inline csharp/HW.02.Task1/Shape.cs]: delete empty

File: csharp/HW.02.Task1/Program.cs [hunk 11]
```diff
@@ -0,0 +1,70 @@
...
+        default:
+            Console.WriteLine("No shape was found!");
+            break;
+
+
+    }
+
+}
+
+
```
Comment (VadzimPapko) [inline csharp/HW.02.Task1/Program.cs]: format

**Comment by sonyashnik** [comment]
Homework Classes and Interfaces

**Pull request merged**